contracts/contracts/layer_1/clearing/ClearingHoldCreationFacet.sol

```diff
@@ -266,20 +267,22 @@ contract ClearingHoldCreationFacet is
         onlyWithValidExpirationTimestamp(
             _clearingOperationFrom.clearingOperation.expirationTimestamp
         )
-        onlyWithValidExpirationTimestamp(_hold.expirationTimestamp)
-        onlyUnProtectedPartitionsOrWildCardRole
         onlyClearingActivated
         returns (bool success_, uint256 clearingId_)
     {
+        {
+            _checkExpirationTimestamp(_hold.expirationTimestamp);
+            _checkUnProtectedPartitionsOrWildCardRole();
+        }
         address sender = _msgSender();
 
         (success_, clearingId_) = _clearingHoldCreationCreation(
             _clearingOperationFrom.clearingOperation,
             _clearingOperationFrom.from,
-            sender,
-            true,
             _hold,
-            _clearingOperationFrom.operatorData
+            _clearingOperationFrom.operatorData,
+            sender,
+            ThirdPartyType.AUTHORIZED
```

Comment:
Sender is used only 1 time and we are mixing the logic of allowance with clearing. We need to review it.